pkg/apis/provisioning/v1alpha5/requirements.go

```diff
@@ -36,21 +42,21 @@ var (
 	// AllowedLabelDomains are domains that may be restricted, but that is allowed because
 	// they are not used in a context where they may be passed as argument to kubelet.
 	// AllowedLabelDomains are evaluated before RestrictedLabelDomains
-	AllowedLabelDomains = sets.NewString(
+	AllowedLabelDomains = stringsets.NewString(
```

Comment:
Why not use the set object you built here? Does it have the same interface?

Reply:
Actually after our discussion I decided to remove most of the unused interface functions from the set object I created. Limiting the scope of the set to only requirement calculation allows us to avoid dealing with quite a few corner cases. It simplify the codes by quite a lot.